crates/diffbot_lib/src/job/runner.rs

```diff
@@ -1,152 +1,52 @@
-
-    let job_clone = job.clone();
-    let output = tokio::task::spawn_blocking(move || runner(&job_clone)).await;
-
-    println!(
-        "[{}] [{}#{}] [{}] Finished",
-        chrono::Utc::now().to_rfc3339(),
-        job.base.repo.full_name(),
-        job.pull_request,
-        job.check_run.id()
-    );
-
-    //eprintln!("Handling job took {}ms", now.elapsed().as_millis());
-
-    if let Err(e) = output {
-        let fuckup = match e.try_into_panic() {
-            Ok(panic) => match panic.downcast_ref::<&str>() {
-                Some(s) => s.to_string(),
-                None => "*crickets*".to_owned(),
-            },
-            Err(e) => e.to_string(),
-        };
-        eprintln!("Join Handle error: {}", fuckup);
-        let _ = job.check_run.mark_failed(&fuckup).await;
-        return;
-    }
-
-    let output = output.unwrap();
-    if let Err(e) = output {
-        let fuckup = format!("{:?}", e);
-        eprintln!("Other rendering error: {}", fuckup);
-        let _ = job.check_run.mark_failed(&fuckup).await;
-        return;
-    }
-
-    match output.unwrap() {
-        CheckOutputs::One(output) => {
-            let res = job.check_run.mark_succeeded(output).await;
-            if res.is_err() {
-                let _ = job
-                    .check_run
-                    .mark_failed(&format!("Failed to upload job output: {:?}", res))
-                    .await;
-            }
+pub async fn handle_output<S: AsRef<str>>(output: Vec<Output>, check_run: CheckRun, name: S) {
+    match output.len() {
+        0usize => {
+            let _ = check_run.mark_failed("Rendering returned nothing!").await;
         }
-        CheckOutputs::Many(first, rest) => {
-            let count = rest.len() + 1;
-
-            let _ = job
-                .check_run
-                .rename(&format!("{} (1/{})", name.as_ref(), count))
+        1usize => {
+            let res = check_run
+                .mark_succeeded(output.into_iter().next().unwrap())
                 .await;
-            let res = job.check_run.mark_succeeded(first).await;
             if res.is_err() {
-                let _ = job
-                    .check_run
+                let _ = check_run
                     .mark_failed(&format!("Failed to upload job output: {:?}", res))
                     .await;
-                return;
             }
-
-            for (i, overflow) in rest.into_iter().enumerate() {
-                if let Ok(check) = job
-                    .check_run
-                    .duplicate(&format!("{} ({}/{})", name.as_ref(), i + 2, count))
-                    .await
-                {
-                    let res = check.mark_succeeded(overflow).await;
-                    if res.is_err() {
-                        let _ = job
-                            .check_run
-                            .mark_failed(&format!("Failed to upload job output: {:?}", res))
+        }
+        _ => {
+            let len = output.len();

```

Comment:
you can bind to output.len() in the match case, replace `_ => { let len // ...` with `len => `